infrastructure/src/main/java/org/wso2/carbon/testgrid/infrastructure/aws/AWSDeployer.java

```diff
@@ -0,0 +1,190 @@
+     * @param awsSecretVariableName Environment variable name for AWS SECRET KEY.
+     * @throws TestGridInfrastructureException Throws exception when environment variables are not set.
+     */
+    public AWSDeployer(String awsKeyVariableName, String awsSecretVariableName) throws TestGridInfrastructureException {
+        String awsIdentity = EnvVariableUtil.readEnvironmentVariable(awsKeyVariableName);
+        String awsSecret = EnvVariableUtil.readEnvironmentVariable(awsSecretVariableName);
+        if ((awsIdentity == null) || (awsSecret == null)) {
+            String errorMessage = "AWS Credentials must be set as environment variables";
+            throw new TestGridInfrastructureException(errorMessage);
+        }
+    }
+
+    /**
+     * This method initiates creating infrastructure through CLoudFormation.
+     *
+     * @param script       Script object containing the CF details.
+     * @param infraRepoDir Path of TestGrid home location in file system as a String.
+     * @return a Deployment object with created infrastructure details.
+     * @throws InterruptedException            when an Interrupt occurs while waiting for CF result.
+     * @throws IOException                     when an error occurs while reading the cf template file.
+     * @throws TestGridInfrastructureException AWS related error caused by malformed CF template.
+     */
+    public Deployment createInfrastructure(Script script, String infraRepoDir) throws InterruptedException,
+            IOException, TestGridInfrastructureException {
+        String cfName = script.getName();
+        AmazonCloudFormation stackbuilder = AmazonCloudFormationClientBuilder.standard()
+                .withCredentials(new EnvironmentVariableCredentialsProvider())
+                .withRegion(infra.getRegion())
+                .build();
+
+        CreateStackRequest stackRequest = new CreateStackRequest();
+        stackRequest.setStackName(cfName);
+        String file = new String(Files.readAllBytes(Paths.get(infraRepoDir, this.infra.getName(),
+                "AWS", "Scripts", script.getName())));
+        stackRequest.setTemplateBody(file);
+        log.info("Created a CloudFormation Stack with the name :" + stackRequest.getStackName());
+        stackbuilder.createStack(stackRequest);
+        waitForAWSProcess(stackbuilder, cfName);
+        DescribeStacksRequest describeStacksRequest = new DescribeStacksRequest();
+        describeStacksRequest.setStackName(cfName);
+        DescribeStacksResult describeStacksResult = stackbuilder.describeStacks(describeStacksRequest);
+        //Initially only PublicDNS which is exported from a dummy script is saved in the Deployment object.
+        List<Host> hosts = new ArrayList<>();
+        for (Stack st : describeStacksResult.getStacks()) {
+            for (Output output : st.getOutputs()) {
+                if (output.getOutputKey().equals("PublicDNS")) {
```

Comment:
How can we guarantee that the cloud formation script has an output defined? Is there a way to obtain the IP address other than from the outputs